otx/mpa/seg/trainer.py

```diff
@@ -57,7 +57,9 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
         datasets = [build_dataset(cfg.data.train)]
 
         # FIXME: Currently segmentor does not support multi batch evaluation.
-        cfg.data.val_dataloader.samples_per_gpu = 1
+        # For the Self-SL case, there is no val data. So, need to check the 
+        if 'val' in cfg.data:
+            cfg.data.val_dataloader.samples_per_gpu = 1
```

Comment:
Could you explain why this change is needed for seg? I don't know the reason because mmseg just already uses `samples_per_gpu=1` when creating val_dataloader. 

https://github.com/openvinotoolkit/mmsegmentation/blob/ac8e04d73857af3bff4c27df62ac69d1c51ff3ac/mmseg/apis/train.py#L65-L74

Reply:
Since global options for dataloader, such as `samples_per_gpu`, etc., would be deprecated in the future as `mmdet` is on the way, MPA explicitly [construct dataloader config](https://github.com/openvinotoolkit/training_extensions/blob/8b1732239eeedde534f1f36e7d89d8951c4d5bb3/otx/mpa/seg/trainer.py#L104) for each subset right before the training. [The method](https://github.com/openvinotoolkit/training_extensions/blob/8b1732239eeedde534f1f36e7d89d8951c4d5bb3/otx/mpa/stage.py#L309-L344) just pops the global options and assigns them to each dataloader config.

Specifying dataloader config would [override the default](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/apis/train.py#L157-L162) which is 1 for validations's `samples_per_gpu`, MPA should set to `1` explicitly.

Reply:
My changing part is to avoid the `cfg.data.val_dataloader.samples_per_gpu=1`.
And after checking the mmseg, I also think we can remove the line #60-62 for the segmentation.

Reply:
> Specifying dataloader config would [override the default](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/apis/train.py#L157-L162) which is 1 for validations's samples_per_gpu, MPA should set to 1 explicitly.

Then you mean that we need to explicitly let users know by adding this line although the default value is set to 1?

Reply:
@sungmanc This should be as is. Unless, it will build dataloader with `samples_per_gpu` from global dataloader option which might not be `1`.

Reply:
Understood, thanks for kind explanation. what do you think? @sungchul2 

If you also agree, we can merge this PR

Reply:
I agree. Thanks for your kind summary!